tools/test/toolchains/test_toolchains.py

```diff
@@ -40,6 +41,58 @@ def test_arm_minimal_printf(self):
         self.assertIn("-DMBED_MINIMAL_PRINTF", arm_micro_obj.flags["common"])
         self.assertIn("-DMBED_MINIMAL_PRINTF", arm_c6_obj.flags["common"])
 
+    def test_arm_default_lib(self):
+        """Test that linker flags are correctly added to an instance of ARM."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.supported_c_libs = {"arm": ["small"]}
+        mock_target.default_lib = "small"
```

Comment:
To also cover case insensitivity
```suggestion
        mock_target.default_lib = "sMAll"
```